lib/matplotlib/axes/_axes.py

```diff
@@ -3952,7 +3961,17 @@ def do_patch(xs, ys, **kwargs):
         elif len(widths) != N:
             raise ValueError(datashape_message.format("widths"))
 
-        for pos, width, stats in zip(positions, widths, bxpstats):
+            # capwidth
+
+        if capwidths is None:
+            capwidths = 0.5 * np.array(widths)
```

Comment:
I am just a little concerned about the units here.  Are widths able to be timedelta64 etc?  Why are widths not just made an array above?

Reply:
Since we only need `widths` to be an np-array for the case `capwidths` is `None`, I preffered not to interfere with the rest of the code. So I did not convert `widths` to a np-array. 

Would you prefere to reset `widths` to be an array? Or should we use  `[0.5 * w for w in widths]` instead of `0.5 * np.array(widths)` ?

The units requirements are basically the same as for `widths`. After all, the caps widths were calculated from them before by multiplying with `0.5`.